File: Week_02/G20190343020289/LeetCode_590_0289.js
// 589. N叉树的前序遍历
/**
 * // Definition for a Node.
 * function Node(val, children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */
/**
 * @param {Node} root
 * @return {number[]}
 */
var preorder = function (root) {
    if (!root) {
        return [];
    }
    var temp = [];
    for (let item of root.children) {
        temp.push(...preorder(item))
    }
    return [root.val, ...temp];
};

File: Week_02/G20190343020289/LeetCode_590_0289.go
package G20190343020289

type Node struct {
	val      int
	children []Node
}

//589. N叉树的前序遍历
func preorder(root *Node) []int {
	if root.children == nil {
		return nil
	}
	temp := make([]int, 0)
	for _, v := range root.children {
		temp = append(temp, preorder(&v)...)
	}
	return append([]int{root.val}, temp...)
}
